fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt

```diff
@@ -499,33 +501,50 @@ private enum class Highlight(
     ),
 }
 
+@Suppress("LongMethod")
 @Composable
 private fun ProductRecommendation(
     product: RecommendedProductState.Product,
     onClick: (String, String) -> Unit,
     onImpression: (String) -> Unit,
 ) {
+    val titleContentDescription = headingResource(id = R.string.review_quality_check_ad_title)
+
     Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
         ReviewQualityCheckCard(
             modifier = Modifier
                 .fillMaxWidth()
-                .semantics { heading() }
-                .clickable {
-                    onClick(product.aid, product.productUrl)
-                }
                 .onShown(
                     threshold = PRODUCT_RECOMMENDATION_IMPRESSION_THRESHOLD,
                     settleTime = PRODUCT_RECOMMENDATION_SETTLE_TIME_MS,
                     onVisible = { onImpression(product.aid) },
                 ),
         ) {
-            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth(1f)
+                    .clearAndSetSemantics {
+                        heading()
+                        contentDescription = titleContentDescription
+                    }
+                    .clickable(interactionSource = MutableInteractionSource(), indication = null) {
```

Comment:
Let's do remember { MutableInteractionSource() }  here so the object is not recreated for every recomposition.